5-network/11-websocket/article.md

```diff
@@ -188,201 +188,201 @@ socket.onmessage = (event) => {
-- `1006` -- no way to set such code manually, indicates that the connection was lost (no close frame).
+- `1000` -- بستن پیش‌فرض و عادی (زمانی که `code` نباشد استفاده می‌شود),
+- `1006` -- راهی برای تنظیم این کد به صورت دستی وجود نداشته و نمایانگر از دست رفتن ارتباط هست (no close frame)
 
-There are other codes like:
+کدهای دیگر مثل موارد زیر هم وجود دارند:
 
 - `1001` -- the party is going away, e.g. server is shutting down, or a browser leaves the page,
-- `1009` -- the message is too big to process,
-- `1011` -- unexpected error on server,
-- ...and so on.
+- `1009` -- حجم پیام برای انجام پردازش زیاد است,
+- `1011` -- خطای پیش‌بینی نشده در سرور,
+- ...و غیره.
 
-The full list can be found in [RFC6455, §7.4.1](https://tools.ietf.org/html/rfc6455#section-7.4.1).
+لیست کامل رو میتونید در [RFC6455, §7.4.1](https://tools.ietf.org/html/rfc6455#section-7.4.1) پیدا کنید.
 
-WebSocket codes are somewhat like HTTP codes, but different. In particular, codes lower than `1000` are reserved, there'll be an error if we try to set such a code.
+کدهای وب سوکت تاحدی مشابه کدهای HTTP میباشند اما متفاوتند. به صورت خاص کدهای کمتر از `1000` از قبل رزرو شده اند و اگر تلاش کنیم تا یکی از این کدهارو استفاده کنیم به ارور برخورد خواهیم کرد.
 
 ```js
-// in case connection is broken
-socket.onclose = event => {
+// اگر ارتباط دچار مشکل باشد:
+socket.onclose = (event) => {
   // event.code === 1006
   // event.reason === ""
   // event.wasClean === false (no closing frame)
 };
 ```
 
+## وضعیت اتصال
 
-## Connection state
+برای اطلاع از وضعیت اتصال پراپرتی `socket.readyState` با مقادیر زیر وجود دارد:
 
-To get connection state, additionally there's `socket.readyState` property with values:
+- **`0`** -- "CONNECTING": اتصال هنوز برقرار نشده است,
+- **`1`** -- "OPEN": درحال برقراری ارتباط,
+- **`2`** -- "CLOSING": درحال بستن ارتباط,
+- **`3`** -- "CLOSED": ارتباط بسته شده است.
 
-- **`0`** -- "CONNECTING": the connection has not yet been established,
-- **`1`** -- "OPEN": communicating,
-- **`2`** -- "CLOSING": the connection is closing,
-- **`3`** -- "CLOSED": the connection is closed.
+## مثال چت
 
+بیاید تا با هم یک مثال از پیاده‌سازی یک برنامه چت را بااستفاده از ای پی آی وب سوکت و ماژول وب سوکت node.js <https://github.com/websockets/ws> بررسی کنیم. تمرکز اصلی ما سمت کلاینت خواهد بود اما سمت سرور هم ساده است.
 
-## Chat example
-
-Let's review a chat example using browser WebSocket API and Node.js WebSocket module <https://github.com/websockets/ws>. We'll pay the main attention to the client side, but the server is also simple.
-
-HTML: we need a `<form>` to send messages and a `<div>` for incoming messages:
+کد HTML: نیاز به یک تگ `<form>` برای ارسال پیامها و یک تگ `<div>` برای پیامهای دریافتی داریم
 
 ```html
 <!-- message form -->
 <form name="publish">
-  <input type="text" name="message">
-  <input type="submit" value="Send">
+  <input type="text" name="message" />
+  <input type="submit" value="Send" />
```

Comment:
Please disable formatting for translations if you're using any. don't change code styles.